scripts/run-reset-hammer.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+# Copyright (c) 2025 Tenstorrent AI ULC
+# SPDX-License-Identifier: Apache-2.0
+
+BOARD=$1
+COUNT=$2
+
+if [ $# -ne 2 ]; then
+	echo "Incorrect number of arguments, expected board name and test count"
+	exit 1
+fi
+
+# Repeatedly resets the SMC. Intended to be run within CI.
+# Accepts two arguments:
+# 1. board name to use
+# 2. the number of times to run the reset test
+
+TT_Z_P="$(dirname $(realpath $0))/.."
+echo $TT_Z_P
+
+fail_cnt=0
+pass_cnt=0
+
+function report_results()
+{
+	echo "Reset test ended: $pass_cnt passed, $fail_cnt failed"
+
+	if [ $fail_cnt -ne 0 ]; then
+		echo "Reset test failed with $fail_cnt errors"
+		exit 1
+	else
+		echo "Reset test passed successfully"
+		exit 0
+	fi
+}
```

Comment:
It would be good to add a follow-up PR that reported a success rate in percent.

E.g.

```shell
success_pct="$(echo "$pass_count * 100 /  ($pass_count + $fail_count)")"

echo "Success Rate: $success_pct %"
```